code/_helpers/lists.dm

```diff
@@ -800,6 +800,60 @@ var/global/list/json_cache = list()
 			log_error("Exception during JSON decoding ([json_to_decode]): [e]")
 	return list()
 
+/proc/load_text_from_directory(var/directory, var/expected_extension = ".txt", var/recursive = TRUE)
+
+	var/list/walked_directories = list()
+	var/list/loaded_files = list()
+	var/dir_count  = 0
+	var/item_count = 0
+
+	// Use a while loop so we can recurse over subdirectories.
+	var/list/directories_to_check = list(directory)
+	// Cache the extension length so we can save some time when checking for matches.
+	expected_extension = trim(lowertext(expected_extension))
+	var/expected_extension_length = -(length(expected_extension))
+	while(length(directories_to_check))
+
+		// Check each directory only once to avoid wasted effort.
+		var/checkdir = directories_to_check[1]
+		directories_to_check -= checkdir
+		walked_directories += checkdir
+
+		// Skip an examples directory if provided.
+		if(checkdir == "[directory]examples/")
```

Comment:
Noting that it should probably be enforced or checked somewhere that `directory` ends in a `/`, or we should make path handling more robust in general.